### ISSUE TITLE
IndexOutOfBounds w/ Latest 'Fix'

### ISSUE THREAD

**Issue by awilliamson** [Funwayguy/InfiniteInvo#29]
Recently updated server and clients to use 1.0.33 on SkyFactory2. When accessing Big Reactors display & control rods server will throw IndexOutOfBounds.

```
[19:11:17] [Server thread/ERROR] [FML]: There was a critical exception handling a packet on channel I_INVO_CHAN
java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
	at java.util.ArrayList.rangeCheck(ArrayList.java:635) ~[?:1.7.0_75]
	at java.util.ArrayList.get(ArrayList.java:411) ~[?:1.7.0_75]
	at infiniteinvo.network.InvoPacket$HandleServer.onMessage(InvoPacket.java:189) ~[InvoPacket$HandleServer.class:?]
	at infiniteinvo.network.InvoPacket$HandleServer.onMessage(InvoPacket.java:49) ~[InvoPacket$HandleServer.class:?]
	at cpw.mods.fml.common.network.simpleimpl.SimpleChannelHandlerWrapper.channelRead0(SimpleChannelHandlerWrapper.java:37) ~[SimpleChannelHandlerWrapper.class:?]
	at cpw.mods.fml.common.network.simpleimpl.SimpleChannelHandlerWrapper.channelRead0(SimpleChannelHandlerWrapper.java:17) ~[SimpleChannelHandlerWrapper.class:?]
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:98) ~[SimpleChannelInboundHandler.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:337) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:323) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[MessageToMessageDecoder.class:?]
	at io.netty.handler.codec.MessageToMessageCodec.channelRead(MessageToMessageCodec.java:111) ~[MessageToMessageCodec.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:337) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:323) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:785) ~[DefaultChannelPipeline.class:?]
	at io.netty.channel.embedded.EmbeddedChannel.writeInbound(EmbeddedChannel.java:169) ~[EmbeddedChannel.class:?]
	at cpw.mods.fml.common.network.internal.FMLProxyPacket.func_148833_a(FMLProxyPacket.java:77) [FMLProxyPacket.class:?]
	at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:212) [ej.class:?]
	at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:165) [nc.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:659) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:334) [lt.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:547) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:427) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer$2.run(MinecraftServer.java:685) [li.class:?]
[19:11:17] [Server thread/INFO]: RadonGaming lost connection: TextComponent{text='A fatal error has occured, this connection is terminated', siblings=[], style=Style{hasParent=false, color=null, bold=null, italic=null, underlined=null, obfuscated=null, clickEvent=null, hoverEvent=null}}
[19:11:17] [Server thread/INFO]: RadonGaming left the game
```

Taken straight from the server console. Let me know if you need any further debug information, I will try to provide.

Thanks to @Xandaros and @JordanTowle for help in finding this.

@Xandaros just found out that issuing a craft command in AE2 also produces this bug.

**Comment by Funwayguy** [comment]
I'll go figure out what broke again this time...

**Comment by awilliamson** [comment]
Just to add information. Server has 3 people. Nobody owns all slots, we all own differing amounts of slots. We upgrade from 1.0.24 -> 1.0.33. We're using a modified SkyFactory2, with added Big Reactors ( latest ) and Draconic Evolution ( latest ).

**Comment by Funwayguy** [comment]
Just posted build 1.0.34 which not only should fix the cause of this crash but should also catch the error if it somehow returns

**Comment by awilliamson** [comment]
Fixed by 82cc9a4605
